classes/user/Report.inc.php

```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ * @defgroup lib_pkp_classes_user
+ */
+
+/**
+ * @file lib/pkp/classes/user/Report.inc.php
```

Comment:
I'm still thinking this abstraction is in the wrong place or at the wrong time. The only current calling code is the user report, where a mapping via
```
switch ($columnName) {
    case 'family_name': return $user->getFamilyName($locale);
    ...
}
```
...would take care of this concisely. If it's an abstraction that is intended for use elsewhere, I'd rather not introduce it until it's been generalized (out of `\PKP\User\Report`) and called from a couple of places to establish a new pattern. Adding abstractions with single calling cases will lead to a profusion of these, which I'm sensitive to as we have a long history of that :)

For now I'd recommend sticking with a more compact implementation, and bringing the mapping stuff to the optimization group for some experimentation beyond the context of the user report -- I think @NateWr is interested in this for the API, for example.

Reply:
Yes, the mappings overlap a lot with the `UserService::getProperties()` method. For now I'd suggest extending `getProperties` as needed. And in the future we can convert the `getProperties` pattern to a class-based mapping, similar to how you've done it here, and roll that out to all of the entities.

Reply:
@asmecher FYI I've ported this code from a plugin, and we had the need to extend it with some custom user data (e.g. the enabled notifications + custom fields), I'll take a look in the comments later. But I think it's not a big deal, as we can still use the plugin :)

Reply:
You should be able to pass into `getProperties` the set of properties you want, and hook into the method to provide values for any properties that are not handled in the core application.

Reply:
@jonasraoni, just checking -- is this still underway? We're hoping to wrap up the release candidate tomorrow.

Reply:
@asmecher 😵 I've been busy with other things, but I'll give it some time today!

Reply:
I've totally removed the mappings, now there's a `_getHeadings` and a `_getDataRow` methods, I could join them in one method, which would look somehow more organized (`return [__('common.id') => $user->getId(), etc];`), but it would be a bit sad to use array_keys/array_values to fill the headings/rows 🤔